File: tests_end_to_end/page_objects/DatasetsPage.py
from playwright.sync_api import Page, expect
from .BasePage import BasePage


class DatasetsPage(BasePage):
    def __init__(self, page: Page):
        super().__init__(page, "datasets")

    def create_dataset_by_name(self, dataset_name: str):
        self.page.get_by_role("button", name="Create new dataset").first.click()
        self.page.get_by_placeholder("Dataset name").fill(dataset_name)
        self.page.get_by_role("button", name="Create dataset").click()

    def select_database_by_name(self, name):
        self.page.get_by_text(name, exact=True).first.click()

    def search_dataset(self, dataset_name):
        self.page.get_by_test_id("search-input").click()
        self.page.get_by_test_id("search-input").fill(dataset_name)

    def check_dataset_exists_on_page_by_name(self, dataset_name):
        expect(self.page.get_by_text(dataset_name).first).to_be_visible()

    def check_dataset_not_exists_on_page_by_name(self, dataset_name):
        expect(self.page.get_by_text(dataset_name).first).not_to_be_visible()

    def delete_dataset_by_name(self, dataset_name):
        self.search_dataset(dataset_name)
        row = (
            self.page.get_by_role("row")
            .filter(has_text=dataset_name)
            .filter(has=self.page.get_by_role("cell", name=dataset_name, exact=True))
        )
        row.get_by_role("button").click()
        self.page.get_by_role("menuitem", name="Delete").click()
        self.page.get_by_role("button", name="Delete dataset").click()

File: tests_end_to_end/typescript-tests/page-objects/datasets.page.ts
import { Page, expect } from '@playwright/test';
import { BasePage } from './base.page';

export class DatasetsPage extends BasePage {
  constructor(page: Page) {
    super(page, 'test-suites');
  }

  async createDatasetByName(datasetName: string): Promise<void> {
    await this.page.getByRole('button', { name: 'Create new' }).first().click();
    await this.page.getByRole('dialog').getByPlaceholder('Name').fill(datasetName);
    await this.page.getByRole('dialog').getByRole('button', { name: 'Create new' }).click();
  }

  async selectDatasetByName(name: string): Promise<void> {
    await this.page.getByRole('cell', { name, exact: true }).click();
    // Wait for navigation to dataset items page
    await this.page.waitForURL(/\/items/);
  }

  async searchDataset(datasetName: string): Promise<void> {
    await this.page.getByTestId('search-input').click();
    await this.page.getByTestId('search-input').fill(datasetName);
  }

  async checkDatasetExists(datasetName: string): Promise<void> {
    await expect(this.page.getByText(datasetName).first()).toBeVisible();
  }

  async checkDatasetNotExists(datasetName: string): Promise<void> {
    await expect(this.page.getByText(datasetName).first()).not.toBeVisible();
  }

  async deleteDatasetByName(datasetName: string): Promise<void> {
    await this.searchDataset(datasetName);

    const row = this.page
      .getByRole('row')
      .filter({ hasText: datasetName })
      .filter({ has: this.page.getByRole('cell', { name: datasetName, exact: true }) });

    await row.getByRole('button').click();
    await this.page.getByRole('menuitem', { name: 'Delete' }).click();
    await this.page.getByRole('button', { name: 'Delete dataset' }).click();
  }
}
